File: chess-game/chess-game/Model/Board/Cell.py
# entity

from Model.Kernel.Entity import Entity
from Model.Pieces.Piece import Piece
from Model.ValueObjects.Color import Color
from Model.ValueObjects.Position import Position

class Cell(Entity):
    def __init__(self, position: Position, color: Color):
        self._position = position
        self._color = color
        self._piece = None
        
    def occupy(self, piece: Piece):
        if not self._piece:
            self._piece = piece
    
    def is_occupied(self) -> bool:
        return self._piece != None

File: chess-game/frontend/src/app/pages/game/play/play/models/board/ cell.ts
import { Piece } from "../pieces/piece";

export class Cell {

    private _file: string;
    private _rank: number;
    private _piece: Piece | null;
    private _color: string;

    constructor(file: string, rank: number, color: string, piece: Piece | null) {
        this._file = file;
        this._rank = rank;
        this._color = color;
        this._piece = piece;
    }

    get file() {
        return this._file;
    }

    get rank() {
        return this._rank;
    }

    get color() {
        return this._color;
    }

    get piece() {
        return this._piece;
    }

    set piece(value: Piece | null) {
        this._piece = value;
    }

    get id() {
        return `${this._file}${this._rank}`.toLowerCase();
    }
}